addons/account/wizard/account_payment_register.py

```diff
@@ -621,7 +621,11 @@ def _reconcile_payments(self, to_process, edit_mode=False):
                                             to which a payment will be created (see '_get_batches').
         :param edit_mode:   Is the wizard in edition mode.
         """
-        domain = [('account_internal_type', 'in', ('receivable', 'payable')), ('reconciled', '=', False)]
+        domain = [
+            ('move_id.state', '=', 'posted'),
```

Comment:
Isn't this the same? (and not accessing another record)
```suggestion
            ('state', '=', 'posted'),
```

Reply:
There is no `state` field on `account.move.line`

Reply:
Indeed, I'm dumb. I thought it was on account_payment but it doesn't make sense regarding the 2 next lines. :upside_down_face: 

Anyway, it still valid if you do `parent_state` (And this one exist on aml :D )
```suggestion
            ('parent_state', '=', 'posted'),
```

Reply:
hmm, maybe this one is relevant https://github.com/odoo/odoo/pull/88842 :)